lib/room.cpp

```diff
@@ -2456,8 +2456,8 @@ void Room::unban(const QString& userId)
 
 void Room::redactEvent(const QString& eventId, const QString& reason)
 {
-    connection()->callApi<RedactEventJob>(id(), QUrl::toPercentEncoding(eventId),
-                                          connection()->generateTxnId(), reason);
+    connection()->callApi<RedactEventJob>(id(), QString::fromUtf8(QUrl::toPercentEncoding(eventId)),
```

Comment:
Upfront percent-encoding of request parameters is deprecated since libQuotient 0.7 - `toPercentEncoding()` shouldn't even have been here.
```suggestion
    connection()->callApi<RedactEventJob>(id(), eventId),
```